spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskMetrics.java

```diff
@@ -55,74 +63,80 @@
 
 	/**
 	 * task status tag. Can be either STATUS_SUCCESS or STATUS_FAILURE
 	 */
-	public static final String TASK_STATUS_TAG = "task.status";
+	public static final String TASK_STATUS_TAG = "spring.cloud.task.status";
 
 	/**
 	 * task exception tag. Contains the name of the exception class in case of error or
 	 * none otherwise.
 	 */
-	public static final String TASK_EXCEPTION_TAG = "task.exception";
-
-	private Timer.Sample taskSample;
+	public static final String TASK_EXCEPTION_TAG = "spring.cloud.task.exception";
 
 	private LongTaskTimer.Sample longTaskSample;
 
-	private Throwable exception;
+	private Observation.Scope scope;
+
+	private TaskExecutionTagsProvider tagsProvider = new DefaultTaskExecutionTagsProvider();
+
+	private TaskExecutionObservationContext taskObservationContext;
 
 	public void onTaskStartup(TaskExecution taskExecution) {
-		LongTaskTimer longTaskTimer = LongTaskTimer
+		if (this.registry != null) {
+			this.taskObservationContext = new TaskExecutionObservationContext(taskExecution);
+			Observation observation = Observation.start(SPRING_CLOUD_TASK_METER, this.taskObservationContext, this.observationRegistry)
+				.contextualName(String.valueOf(taskExecution.getExecutionId()))
+				.tagsProvider(this.tagsProvider);
+			this.scope = observation.openScope();
+			LongTaskTimer longTaskTimer = LongTaskTimer
 				.builder(SPRING_CLOUD_TASK_ACTIVE_METER).description("Long task duration")
-				.tags(commonTags(taskExecution)).register(Metrics.globalRegistry);
+				.tags(commonTags(taskExecution)).register(this.registry);
 
-		this.longTaskSample = longTaskTimer.start();
-		this.taskSample = Timer.start(Metrics.globalRegistry);
+			this.longTaskSample = longTaskTimer.start();
+		}
 	}
 
 	public void onTaskFailed(Throwable throwable) {
-		this.exception = throwable;
+		if (this.registry != null) {
+			this.taskObservationContext.setExceptionMessage(throwable.getClass().getSimpleName());
+			this.taskObservationContext.setStatus(STATUS_FAILURE);
+		}
 	}
 
 	public void onTaskEnd(TaskExecution taskExecution) {
-		if (this.taskSample != null) {
-			this.taskSample.stop(Timer.builder(SPRING_CLOUD_TASK_METER)
-					.description("Task duration").tags(commonTags(taskExecution))
-					.tag(TASK_EXIT_CODE_TAG, "" + taskExecution.getExitCode())
-					.tag(TASK_EXCEPTION_TAG,
-							(this.exception == null) ? "none"
-									: this.exception.getClass().getSimpleName())
-					.tag(TASK_STATUS_TAG,
-							(this.exception == null) ? STATUS_SUCCESS : STATUS_FAILURE).register(Metrics.globalRegistry));
-			this.taskSample = null;
-		}
-
-		if (this.longTaskSample != null) {
-			this.longTaskSample.stop();
-			this.longTaskSample = null;
+		if (this.scope != null) {
+			this.taskObservationContext.getTaskExecution().setExitCode(taskExecution.getExitCode());
+			this.scope.getCurrentObservation().stop();
+			this.observationRegistry.getCurrentObservation().stop();
+			this.observationRegistry.setCurrentObservation(null);
```

Comment:
Instead of 

```java
this.scope.getCurrentObservation().stop();
			this.observationRegistry.getCurrentObservation().stop();
			this.observationRegistry.setCurrentObservation(null);
```

you can do 

```java
this.scope.getCurrentObservation().stop();
this.scope.close();
```